src/renderer/components/inputs/SliderInput.tsx

```diff
@@ -57,17 +58,70 @@ const SliderInput = memo(
         isLocked,
     }: SliderInputProps) => {
         const [input, setInput] = useInputData<number>(index);
+        const [inputString, setInputString] = useState(String(input));
         const [sliderValue, setSliderValue] = useState(input ?? def);
         const [showTooltip, setShowTooltip] = useState(false);
 
-        const formatValString = (val: string) => `${val}${unit ?? ''}`;
+        const precision = Math.max(getPrecision(offset), getPrecision(step));
+        const addUnit = (val: string) => `${val}${unit ?? ''}`;
+        const dynamicNumInputWidth = 3 + 0.5 * precision + 0.5 * (unit?.length ?? 0);
 
         useEffect(() => {
             setSliderValue(input);
+            setInputString(String(input));
         }, [input]);
 
+        const onSliderChange = (sliderInput: number) => {
+            setInputString(String(sliderInput));
+            setSliderValue(sliderInput);
+        };
+
+        const onNumberInputChange = (numberAsString: string) => {
+            setInputString(numberAsString);
+            setSliderValue(Number(numberAsString));
+        };
+
+        const onBlur = () => {
```

Comment:
This is just a copy of the onBlur logic from inputs/NumberInput, right?

Would it make sense to move all the logic around Chakra's `<NumberInput>` element into its own component?

Reply:
It is 99% similar. The only difference is something that I can actually take out now anyway in NumberInput because of the useEffect you added I believe.

Reply:
There are actually some differences in how the NumberInput field is formatted in SliderInput, after looking at it again once I knew what you were describing (size, stepper m and p, input field width, etc.). This *could* I think be gotten around by adding extra inputs to the NumericalInput function after separating it out, but then these would need to be added to the Props interface as well. So I'm thinking leave it as is for this release, and refactor it later if we want.